### PR TITLE
Support unity code coverage package.

### PR DESCRIPTION
**Pull request by Fehrox**
First off; awesome project we've found it super useful. Thanks for building, sharing and maintaining it so well!

Unity now has a great [code coverage package](https://docs.unity3d.com/Packages/com.unity.testtools.codecoverage@0.2/manual/CoverageTestRunner.html) now available and we're big on testing at AIR so we're using it. Naturally we want to integrate this as a quality gate with our existing github actions, which makes use of your project. To do that we've added the new code coverage arguments to the unity arguments as you can see in our amendments to [run_tests.sh](https://github.com/webbertakken/unity-test-runner/compare/master...AnImaginedReality:master#diff-2abcc3f049b9f46d949f8ef1f5849c4d).

It's worth noting that we've tested this with a project that doesn't use the above package, and the arguments where simply ignored.

Once the artifacts contain the code coverage report, we then parse it with our own github action job, found here: https://github.com/AnImaginedReality/UnityCodeCoverage.Action.

We'll probably continue to improve and maintain the UnityCodeCoverage.Action repo, and wanted to thank you for your great work building and maintaining this project so far.

### CODE REVIEW

File: action/steps/run_tests.sh [hunk 1]
```diff
@@ -82,6 +82,10 @@ if [ $EDIT_MODE = true ]; then
       -runTests \
       -testPlatform editmode \
       -testResults "$FULL_ARTIFACTS_PATH/editmode-results.xml" \
+      -enableCodeCoverage \
+      -coverageResultsPath "$FULL_ARTIFACTS_PATH/CodeCoverage/" \ 
+      -coverageOptions "generateHtmlReport;assemblyFilters:-*unity*" \
+      -burst-disable-compilation
```
Comment (webbertakken) [inline action/steps/run_tests.sh:88]: You're missing a backslash at the end of this line

File: action/steps/run_tests.sh [hunk 2]
```diff
@@ -120,6 +124,10 @@ if [ $PLAY_MODE = true ]; then
       -runTests \
       -testPlatform playmode \
       -testResults "$FULL_ARTIFACTS_PATH/playmode-results.xml" \
+      -enableCodeCoverage \
+      -coverageResultsPath "$FULL_ARTIFACTS_PATH/CodeCoverage/" \
+      -coverageOptions "generateHtmlReport;assemblyFilters:-*unity*" \
+      -burst-disable-compilation
```
Comment (webbertakken) [inline action/steps/run_tests.sh:130]: same here

**Comment by webbertakken** [comment]
Hey, thanks for your PR. I think it's a nice step forward and I'm open to merging it.

The failing tests may be related to https://github.com/webbertakken/unity-test-runner/issues/54

I haven't had time to investigate this thoroughly.

File: action/steps/run_tests.sh (hunk 1 above)
Comment (Fehrox) [reply action/steps/run_tests.sh]: Amended.

File: action/steps/run_tests.sh (hunk 2 above)
Comment (Fehrox) [reply action/steps/run_tests.sh]: Amended.

**Comment by Fehrox** [comment]
Yeah it's quite strange, the actions execute just fine as part of our build pipeline. Could be something with the text encoding of run_tests.sh, though I can't make sense of why it'd run inconsistently, especially in a container. 

I'll look into it when I get a bit of time.

**Comment by webbertakken** [comment]
LGTM. 

I'm sorry but I would prefer to indeed fix that issue first before merging anything. It's probably not related to your MR directly though.

**Comment by Fehrox** [comment]
FYI, I did revert that last amendment, as it caused errors with our build pipeline including the '\' at the end of the command.

**Comment by Fehrox** [comment]
> LGTM.
> 
> I'm sorry but I would prefer to indeed fix that issue first before merging anything. It's probably not related to your PR directly though.

Yeah that's absolutely fine. I like this project and where it's heading so I'll take the time to help with that open issue when I have some. I agree also that it should be resolved before the pull request is accepted.

**Comment by webbertakken** [comment]
Looking at this again, I think it's not correctly implemented. Perhaps there's a package missing?

The failure does not seem related to the issue i linked before.

**Comment by ivan-omnidrone** [comment]
Probably worth noting that running test coverage will slow down the unit tests, according to documentation [here](https://docs.unity3d.com/Packages/com.unity.testtools.codecoverage@0.2/manual/UsingCodeCoverage.html).
Shouldn't be possible to achieve the same goal using customParameters?

**Comment by webbertakken** [comment]
> Probably worth noting that running test coverage will slow down the unit tests, according to documentation [here](https://docs.unity3d.com/Packages/com.unity.testtools.codecoverage@0.2/manual/UsingCodeCoverage.html).
> Shouldn't be possible to achieve the same goal using customParameters?

It should be possible yes, though it would be easier if users that are implementing their pipeline don't have to think about the exact parameters unity needs, and instead talk against a different abstraction. (like we have now; you only need to know how the action works, not how unity command line works)

**Comment by webbertakken** [comment]
As the de facto way to test has not fully been settled and the related package is still in early stages there are some doubts about this PR.

Slowing down the tests while not having it configurable as an option like `coverage: true` is kind of tricky and may endanger backward compatiblity.

Closing this as there hasn't been much ask for it and the discussion stagnated.

**Comment by ysma500** [comment]
I'd be renewing interest on this feature 💯

**Comment by GabLeRoux** [comment]
For refence, this is how it was implemented in the gitlab-ci example project:
https://gitlab.com/gableroux/unity3d-gitlab-ci-example/-/blob/e2adfbb1/.gitlab-ci.yml#L61
https://gitlab.com/gableroux/unity3d-gitlab-ci-example/-/blob/master/ci/test.sh#L18
https://gitlab.com/gableroux/unity3d-gitlab-ci-example/-/blob/e2adfbb1/ci/test.sh#L35-43

**Comment by sbergen** [comment]
Just a note here, also related to execution speed: To get accurate coverage numbers with Unity >= 2020, you need use the `-debugCodeOptimization` parameter.

**Pull request closed without merging**